server/http/service.go

```diff
@@ -33,5 +33,6 @@ func NewService(router *mux.Router, apps *apps.Service, initf ...func(*mux.Route
 
 // Handle should be called by the plugin when a command invocation is received from the Mattermost server.
 func (s *service) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	r.Header.Add("MM_SESSION_ID", c.SessionId)
```

Comment:
I 👎  Gorilla for this - no easy way to pass extra data into the handlers. They recommend using `Context` but 3/5 it's a wrong thing to do.